### PR TITLE
feat: simulation database integration and create APIs

### PR DESCRIPTION
**Pull request by ReebaPaysys**
# SPDX-License-Identifier: Apache-2.0
This pull request adds comprehensive unit tests for several core service and repository modules, significantly improving test coverage and reliability for simulation suites, node logic, and data model operations. The tests use Jest to mock dependencies and validate behaviors, error handling, and edge cases for both service and repository layers.

**Simulation Suites API Handlers and Logic:**

* Adds detailed tests for simulation suite handlers in `app.controller`, covering creation, retrieval (with pagination and filtering), updating, and error delegation to `ErrorHandler`. Tests validate correct mapping of request data, status codes, and error handling logic.

**Node Logic and Repository:**

* Adds unit tests for `node.logic.service`, verifying node creation (including duplicate name conflict and normalization of payload), retrieval, deletion, filtering, and secure query execution with tenant isolation. Tests also ensure only SELECT queries are allowed and tenant scoping is properly injected.
* Adds repository-level tests for `node.repository`, covering node retrieval by name and ID, insertion, deletion, sorting, and direct query execution. Tests validate SQL statement construction, parameter usage, and fallback behavior for invalid sort fields.

**Data Model Repository:**

* Introduces tests for `data-model.repository`, ensuring correct retrieval and upsert of data model JSON per tenant, including error handling when upsert fails.
## How was it tested?
- [ ] Locally
- [ ] Development Environment
- [ ] Not needed, changes very basic
- [ ] Husky successfully run
- [ ] Unit tests passing and Documentation done

### CODE REVIEW

**Comment by ReebaPaysys** [comment]
### **Lint**
image

### **Test Case**

image

File: src/router.ts [hunk 1]
```diff
@@ -446,6 +454,24 @@ function Routes(fastify: FastifyInstance): void {
   fastify.get('/v1/admin/simulation-logs/:ruleId', {
     ...SetOptionsBodyAndParams(getSimulationLogsHandler, routePrivilege.getSimulationLogs),
   });
+
+  // Simulation Studio
```
Comment (ahmad-paysys) [inline src/router.ts:458]: Add typed Fastify schemas for body/query shapes here for all simulation studio endpoints.

File: src/services/simulation-suites.logic.service.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  userId: string,
+  userEmail?: string,
+): Promise<SimulationSuite> => {
+  try {
+    // Validate required fields
+    if (!payload.name || payload.name.trim().length === 0) {
+      throw new HttpException('Simulation suite name is required', HttpStatus.BAD_REQUEST);
+    }
+
+    if (payload.name.length > 30) {
```
Comment (ahmad-paysys) [inline src/services/simulation-suites.logic.service.ts]: Please make sure `payload.name.length` validation is consistent across the create and update endpoints. Right now, the create endpoint enforces a 30-character limit, while the [patch/update endpoint allows 50 characters](https://github.com/tazama-lf/admin-service/pull/374/changes#diff-9cfe200dc96689f1f8573a4a418f6dc5cfb570348dc8540963fe5266ffbd66ffR123).

Please standardize this to 50 characters across both endpoints. Also, avoid using magic numbers directly in validation logic — define a shared constant for limits like this. Otherwise, it becomes easy for these constraints to drift out of sync over time when changes are made in different parts of the codebase.
